lib/PhasarLLVM/TypeHierarchy/DIBasedTypeHierarchy.cpp

```diff
@@ -125,47 +122,80 @@ DIBasedTypeHierarchy::DIBasedTypeHierarchy(const LLVMProjectIRDB &IRDB) {
     TransitiveClosure[I][I] = true;
   }
 
-  // get VTables
-  const auto Subprograms = Finder.subprograms();
-  for (auto *Subprogram : Subprograms) {
+  size_t VTableSize = 0;
+  for (const auto &Subprogram : Finder.subprograms()) {
+    if (const auto *SubProgramType =
+            llvm::dyn_cast<llvm::DISubprogram>(Subprogram)) {
+      if (SubProgramType->getVirtualIndex() > VTableSize) {
+        VTableSize = SubProgramType->getVirtualIndex();
+      }
+    }
+  }
+  // if the biggest virtual index is 2 for example, the vector needs to have a
+  // size of 3 (Indices: 0, 1, 2)
+  VTableSize++;
+
+  std::vector<const llvm::Function *> Init = {};
+  std::vector<std::vector<const llvm::Function *>> IndexToFunctions;
+  for (size_t I = 0; I < VTableSize; I++) {
+    IndexToFunctions.push_back(Init);
+  }
```

Comment:
I don't think, the size of IndexToFunctions is tied to the max virtual index; rather it is the number of types. The sizes of the inner vectors depend on that, but on a per-type basis